src/openrct2/ride/Vehicle.cpp

```diff
@@ -7479,8 +7479,8 @@ void Vehicle::UpdateSceneryDoor() const
     auto wallCoords = CoordsXYZ{ x, y, TrackLocation.z - trackBlock->z + trackCoordinates->z_end }.ToTileStart();
     int32_t direction = (track_direction + trackCoordinates->rotation_end) & 3;
 
-    AnimateSceneryDoor(
-        { wallCoords, static_cast<Direction>(direction) }, TrackLocation, next_vehicle_on_train == SPRITE_INDEX_NULL, false);
+    AnimateSceneryDoor<false>(
```

Comment:
What does this solve exactly? It still acts exactly like a regular parameter.

Reply:
Since the parameter is always constant, there is no reason for passing it as a function parameter.

Reply:
This allows the compiler to do better optimizations.
I can remove it if you want 😃

Reply:
> This allows the compiler to do better optimizations.

Now, _that_ is a proper reason. Let's leave it like this, then.

Reply:
We've done similar in the software drawing functions. I don't think in this case it's really required but I don't mind it.